### PR TITLE
Migration to JBake

### PR DESCRIPTION
**Pull request by tarilabs**
## Context and Problem statement

jugmilano.it is a static website, currently generated with Jekyll and customized templates.
Sometimes, modifications/contributions are needed from folks not familiar with Ruby/Jekyll.
Faced challenges:
- quick corrections of the YAML [remotely via smartphone](https://github.com/jugmilano/jugmilano.github.io/commit/d3ff098924ba8961577ed13cdc2f832e0a6047a8) was challenging while on holiday 🏖️ 🍹 
- folks mentioned is sometimes difficult to install Ruby/chruby/jekyll just to test locally
- folks mentioned is sometimes tedius to spin a docker container just to test locally

This proposed changes migrates FROM Jekyll TO [JBake](https://github.com/jbake-org/jbake).

- JBake is Java/JVM based analogous of Jekyll
- Build is automated using Maven (another familiar build tool for JVM devs)

## Goals

- Migrate from Jekyll by avoiding as much as possible modification to _content_ files.
  - achieved by customizing markdown engine processing `.md` files' front-matter ([ref](https://jekyllrb.com/docs/front-matter/))
- Preserve as much git history as possible
  - except of few files, content was rearranged preserving attribution and git history
- Instrument website build via Maven
- CI/CD for publishing on GH Pages via GHA
- Quick utility for "previewing" and testing the website _locally_

## Non-goals

- Additional CI/CD or infra changes (can be addressed in subsequent PR/changes)
- Modification to layout/styles/pages (can be addressed in subsequent PR/changes)

## Testing these changes

Checkout this PR and follow 2 commands in README.md.

### CODE REVIEW

**Review by lucamolteni — APPROVED** [review]
OMG thanks

File: _posts/faq.md [hunk 1]
```diff
@@ -9,36 +9,28 @@ layout: content
...
-<!---
-Usiamo le "sezioni" di Markdown affinché ogni "titolo" corrisponda anche ad un ID nel html renderizzato, in modo da poterlo linkare come #titolo .
--->
-
-### Chi siamo ###
 
 Il JVM User Group _"JUG Milano"_ è un gruppo di appassionati di programmazione che lavorano o gravitano attorno a Java, alla JVM, a tutti i linguaggi e le tecnologie del suo ecosistema. Ma non solo.
 
-### Canali di comunicazione ###
+<h3 id="canali-di-comunicazione">Canali di comunicazione</h3>
```
[the remark this answers is not in the document]
Comment (tarilabs) [reply _posts/faq.md:15]: Ho solo convertito as-is da Header di livello 3 in MD a H3 in HTML per aggiungere un `id` esplicito.

Penso optai per un livello 3 ai tempi per evitare che nella FAQ la domanda sia 

# Troppo grande?
Rispetto alla risposta

[the remark this answers is not in the document]
Comment (lucamolteni) [reply _posts/faq.md:15]: credo che in markdown l'ID sia il titolo stesso (o perlomeno quando fai un link interno si comporta così). A cosa ti serviva l'id esplicito?

[the remark this answers is not in the document]
Comment (lucamolteni) [reply _posts/faq.md:15]: Fine by me

[the remark this answers is not in the document]
Comment (tarilabs) [reply _posts/faq.md]: @lucamolteni vedi se 1b229fc26e47d7e661df72c7cb4e6017b47b91b3 e' accettabile, purtroppo bisogna forzare le estensioni

[the remark this answers is not in the document]
Comment (lucamolteni) [reply _posts/faq.md]: Mi pare perfetto grazie

[the remark this answers is not in the document]
Comment (tarilabs) [reply _posts/faq.md]: Pegdown md profile is used ootb in JBake.

For the records, https://github.com/jugmilano/jugmilano.github.io/commit/1b229fc26e47d7e661df72c7cb4e6017b47b91b3 hardcoded additional extensions, which generate `id=` attribute in rendered HTML.

These extensions are coming from flexmark, beyond Pegdown md profile.

**Pull request merged**